odoo/tools/populate.py

```diff
@@ -0,0 +1,72 @@
+import random
+from odoo.tools import pycompat
+
+def Random(seed):
+    r = random.Random()
+    r.seed(seed, version=2)
+    return r
+
+
+def format_str(val, counter, _): # todo format
+    if isinstance(val, str) and '%' in val:
+        return val % counter
+    return val
+
+def chain_generators(fields_generators):
+    generator = root_generator()
+    for (fname, field_generator) in fields_generators:
+        generator = field_generator(generator, fname)
+    return generator
+
+def root_generator():
+    yield {}, False
+    while True:
+        yield {}, True
+
+
+def randomize(vals, weights=None, seed=False, formater=format_str):
+    def generator(iterator, field_name, counter_offset=0):
+        r = Random(seed or field_name)
+        for counter, (values, complete) in enumerate(iterator):
+            val = r.choices(vals, weights)[0]
```

Comment:
```suggestion
            val = r.choice(vals, weights)
```

Reply:
random.choice doesn't manage weights. but since choices is not available in python2, we may need to create a compatibility. I'll add this feature on the tool populate Random